pkg/virt-operator/creation/components/scc.go

```diff
@@ -0,0 +1,82 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2019 Red Hat, Inc.
+ *
+ */
+package components
+
+import (
+	"fmt"
+
+	secv1 "github.com/openshift/api/security/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func GetAllSCC(namespace string) []*secv1.SecurityContextConstraints {
+	return []*secv1.SecurityContextConstraints{
+		NewKubeVirtHandlerSCC(namespace),
+		NewKubeVirtControllerSCC(namespace),
+	}
+}
+
+func newBlankSCC() *secv1.SecurityContextConstraints {
+	return &secv1.SecurityContextConstraints{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "security.openshift.io/v1",
+			Kind:       "SecurityContextConstraints",
+		},
+	}
+}
+
+func NewKubeVirtHandlerSCC(namespace string) *secv1.SecurityContextConstraints {
+	scc := newBlankSCC()
+
+	scc.Name = "kubevirt-handler"
+	scc.AllowPrivilegedContainer = true
+	scc.AllowHostPID = true
+	scc.AllowHostPorts = true
+	scc.AllowHostIPC = true
+	scc.RunAsUser = secv1.RunAsUserStrategyOptions{
+		Type: secv1.RunAsUserStrategyRunAsAny,
+	}
+	scc.SELinuxContext = secv1.SELinuxContextStrategyOptions{
+		Type: secv1.SELinuxStrategyRunAsAny,
+	}
+	scc.Volumes = []secv1.FSType{secv1.FSTypeAll}
+	scc.AllowHostDirVolumePlugin = true
+	scc.Users = []string{fmt.Sprintf("system:serviceaccount:%s:kubevirt-handler", namespace)}
+
+	return scc
+}
+
+func NewKubeVirtControllerSCC(namespace string) *secv1.SecurityContextConstraints {
+	scc := newBlankSCC()
+
+	scc.Name = "kubevirt-controller"
+	scc.AllowPrivilegedContainer = false
+	scc.RunAsUser = secv1.RunAsUserStrategyOptions{
+		Type: secv1.RunAsUserStrategyRunAsAny,
+	}
+	scc.SELinuxContext = secv1.SELinuxContextStrategyOptions{
+		Type: secv1.SELinuxStrategyRunAsAny,
+	}
+	scc.AllowedCapabilities = []corev1.Capability{secv1.AllowAllCapabilities}
```

Comment:
we know the exact capabilities required. From pkg/virt-controller/services/template.go it should be, NET_ADMIN, SYS_NICE, and SYS_RESOURCE.

Reply:
I had these capabilities before and each functional test failed with `LibvirtError(Code=8, Domain=44, Message='invalid argument: could not find capabilities for arch=x86_64 domaintype=kvm)` ([link to failed tests](https://prow.apps.ovirt.org/view/gcs/kubevirt-prow/pr-logs/pull/kubevirt_kubevirt/2545/pull-kubevirt-e2e-os-3.11.0/1158790334914760706)) 

I was afraid I will have to add many KVMs capabilities that will make the list very long https://www.kernel.org/doc/Documentation/virtual/kvm/api.txt so I decided to allow all because needed capabilities are defined in template.go so anything beyond that would not be added to the pod and the `kubevirt-controller` SCC is assigned only to `kubevirt-controller` service account.

Although, I will check that again.

Reply:
umm, I'm not sure what you are encountering there. This AllowedCapabilities list isn't actually impacting what capabilities the Pod receives. It's just allowing virt-controller to create a pod with those capabilities which is something that is already occurring. 

I wouldn't expect the values in this list to impact the VMI at runtime.

Reply:
you're right, it seems my environment was messed, on fresh cluster it works.